packages/gestalt/src/Masonry/dynamicHeightsV2Utils.ts

```diff
@@ -110,7 +112,7 @@ function recalcHeights<T>({
   if (
     !changedItemPosition ||
     newHeight === 0 ||
-    Math.floor(changedItemPosition.height) === Math.floor(newHeight)
+    Math.floor(Math.abs(newHeight - changedItemPosition.height)) === 0
```

Comment:
This is a different change that should be added on a different PR since it could have unexpected side effects.